doc/specs/#476 - Pinned packages.md

```diff
@@ -0,0 +1,180 @@
+---
+author: Easton Pillay jedieaston/easton@planeteaston.com
+created on: 2022-01-25
+last updated: 2022-02-02
+issue id: 476
+---
+
+# Pinning a package
+
+For [#476](https://github.com/microsoft/winget-cli/issues/476)
+
+## Abstract
+
+This spec describes the functionality behind "pinning" a package (freezing a package at a certain version so that it will be not be automatically upgraded).
+
+## Inspiration
+
+This functionality is inspired by functionality in other package managers, as well as community feedback. Many packages (software development tools and libraries commonly, but other software as well) introduce breaking changes that users may not want integrate into their workflow quite yet.
+
+## Solution Design
+
+A table of packages (by Name, Version and Product Code if available) that are currently pinned will be kept in the local tracking catalog. To be repository independent, these values should be correlated from local sources (Add and Remove Programs, Appx, etc). This also means that the user should be able to pin packages that were not installed via the Windows Package Manager, so that they won't be upgraded via `winget upgrade --all`. 
```

Comment:
With the traditional Microsoft Store packages, the Microsoft Store automatically updates programs. Users would not be able to pin those packages. I need to see if there is a way to pin packages from being automatically updated by the Microsoft Store.

Reply:
I was trying to be as repo neutral as possible, but the Microsoft Store will definitely need special treatment. We need to be able to pin packages even if they are coming from REST sources.

Reply:
At least with chocolatey, I ended up pinning for two reasons:
* I really want a specific version
* The application has it's own update mechanism

So the behavior of "the package may still auto update, and is just excluded from `winget upgrade --all` makes a lot of sense to me.